presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java

```diff
@@ -145,26 +159,28 @@ private void startLoadSplits()
             return;
         }
         if (outstandingTasks.incrementAndGet() > maxPartitionBatchSize) {
-            outstandingTasks.decrementAndGet();
+            if (outstandingTasks.decrementAndGet() == 0) {
+                invokeFinishedIfShould();
```

Comment:
maybe name this `endTask` and move the `if (outstandingTasks.decrementAndGet() == 0)` inside the method